connectd/multiplex.c

```diff
@@ -399,6 +399,66 @@ static struct io_plan *io_sock_shutdown_cb(struct io_conn *conn, struct peer *un
 	return io_sock_shutdown(conn);
 }
 
+/* Process and eat protocol_batch_element messages, encrypt each element message
+ * and return the encrypted messages as one long byte array. */
+static u8 *process_batch_elements(struct peer *peer, const u8 *msg TAKES)
+{
+	u8 *ret = tal_arr(peer, u8, 0);
+	size_t ret_size = 0;
+	const u8 *cursor = msg;
+	size_t plen = tal_count(msg);
+
+	status_debug("Processing batch elements of %zu bytes. %s", plen,
+		     tal_hex(tmpctx, msg));
+
+	do {
+		u8 *element_bytes;
+		u16 element_size;
+		struct channel_id channel_id;
+		u8 *enc_msg;
+
+		if (fromwire_u16(&cursor, &plen) != WIRE_PROTOCOL_BATCH_ELEMENT)
+			status_failed(STATUS_FAIL_PROTO_BATCH,
+				      "process_batch_elements on msg that is"
+				      " not WIRE_PROTOCOL_BATCH_ELEMENT. %s",
+				      tal_hexstr(tmpctx, cursor, plen));
```

Comment:
No: don't kill connectd because a channeld misbehaves.  status_broken() here, return tal_free(ret) and have the caller caller io_close() on the peer in this case.